extensions-core/druid-basic-security/src/main/java/org/apache/druid/security/basic/BasicSecurityDruidModule.java

```diff
@@ -168,6 +211,33 @@ public static BasicAuthorizerCacheNotifier createAuthorizerCacheNotifier(final I
     );
   }
 
+  /**
+   * Returns the instance provided either by a config property or coordinator-run class or default class.
+   * The order of check corresponds to the order of method params.
+   */
+  @Nullable
+  public static <T> T getInstance(
+      Injector injector,
+      String configClassName,
+      Class<? extends T> classRunByCoordinator,
+      Class<? extends T> defaultClass
+  ) throws ClassNotFoundException
+  {
+    if (configClassName != null) {
+      // ClassCastException is thrown in case of a mismatch, configuration fix is required.
+      @SuppressWarnings("unchecked")
+      final T instance = (T) injector.getInstance(Class.forName(configClassName));
+      return instance;
+    }
+    if (classRunByCoordinator != null && isCoordinator(injector)) {
+      return injector.getInstance(classRunByCoordinator);
+    }
+    if (defaultClass != null) {
+      return injector.getInstance(defaultClass);
+    }
+    return null;
```

Comment:
Not sure what does returning `null` mean for such method. Is it a failure? Then why not IllegalArgumentException? In any case, it would be better to avoid returning null in a simpler method.

Reply:
Null is for backward compatibility (for cases when a specific component is not needed)

Reply:
Literally, null is needed only as a return value to satisfy the signature of  `@Provides` methods which won't be called

Reply:
Why not to throw an AssertException then (and log `assertionError()`)?

Reply:
Added the exception